src/common/libflux/fripp.c

```diff
@@ -341,84 +368,107 @@ static void timer_cb (flux_reactor_t *r,
 
 void fripp_set_agg_period (struct fripp_ctx *ctx, double period)
 {
+    if (!fripp_enabled (ctx))
+        return;
+
     if (period <= 0) {
         flux_watcher_stop (ctx->w);
         ctx->period = 0;
         return;
     }
 
     ctx->period = period;
+
     double after = flux_watcher_next_wakeup (ctx->w) - (double) time (NULL);
 
     flux_timer_watcher_reset (ctx->w, after, ctx->period);
 }
 
-void fripp_ctx_destroy (void *arg)
+void fripp_ctx_destroy (struct fripp_ctx *ctx)
 {
-    struct fripp_ctx *ctx = arg;
-
-    if (ctx->w) {
-        flux_watcher_stop (ctx->w);
+    if (ctx) {
+        int saved_errno = errno;
         flux_watcher_destroy (ctx->w);
-    }
-    if (ctx->metrics) {
-        zhashx_purge (ctx->metrics);
-        zhashx_destroy (&ctx->metrics);
-    }
-    if (ctx->done) {
+        if (ctx->metrics) {
+            zhashx_purge (ctx->metrics);
+            zhashx_destroy (&ctx->metrics);
+        }
         zlist_destroy (&ctx->done);
+        if (ctx->sock != -1)
+            close (ctx->sock);
+        free (ctx->buf);
+        if (ctx->addrinfo)
+            freeaddrinfo (ctx->addrinfo);
+        free (ctx);
+        errno = saved_errno;
     }
-    close (ctx->sock);
-    free (ctx->buf);
-    free (ctx);
 }
 
-struct fripp_ctx *fripp_ctx_create (flux_t *h)
+static int parse_address (struct addrinfo **aip,
+                          const char *s,
+                          char *errbuf,
+                          int errbufsz)
 {
-    struct fripp_ctx *ctx;
-    char *uri, *port_s = NULL, *host = NULL;
-
-    if (!(ctx = calloc (1, sizeof (*ctx)))) {
-        flux_log_error (h, "fripp_ctx_create");
-        goto error;
-    }
-    if (!(uri = getenv ("FLUX_FRIPP_STATSD"))) {
-        flux_log_error (h, "FLUX_FRIPP_STATSD env var not set");
-        goto error;
-    }
-    if (!(port_s = strrchr (uri, ':'))) {
-        flux_log_error (h, "FLUX_FRIPP_STATSD env var no port");
-        goto error;
+    char *node;
+    char *service;
+    struct addrinfo hints, *res = NULL;
+    int e;
+
+    if (!(node = strdup (s))) {
+        snprintf (errbuf, errbufsz, "out of memory");
+        return -1;
     }
-    if (!(host = calloc (1, port_s - uri + 1))) {
-        flux_log_error (h, "fripp_ctx_create");
+    if (!(service = strchr (node, ':'))) {
+        snprintf (errbuf, errbufsz, "missing colon delimiter");
```

Comment:
Looks like tests of malformed `FLUX_FRIPP_STATSD` could be added (missing `:` and failure from `getaddrinfo()` below) to give better coverage of these conditions, but that is optional IMO.